### ISSUE TITLE
The build is not run again if I update the pull request

### ISSUE THREAD

**Issue by k33g** [takezoe/gitbucket-ci-plugin#35]
Hello @takezoe 
I created a PR with a nodejs project and tests with mocha
if the test is ko, the build fails
I fix my test, but the PR status stays to ` Waiting for status to be reported` and there is no build

I created a new PR with a "good" test, the build is successful, there is no more build if I update my source code (for the open PR)

**Comment by takezoe** [comment]
@k33g I tested on my environment but couldn't reproduce. Is it occurred always? If yes, let me know detailed procedure to reproduce.

**Comment by k33g** [comment]
@takezoe I will try with an other laptop and write all te procedure

**Comment by k33g** [comment]
Hello @takezoe, here is my procedure:

I'm using:
- gitbucket: 4.17.0
- gitbucket-ci-plugin: 1.0.1

I created a group `ACME`
I created a repository `demo` for the `ACME` group

I add a file: `tests/test.js`

```javascript
'use strict';

var expect = require('chai').expect;

describe('# 42 is 42', () => {
  it('should equal 42', () => {
    expect(42).to.equal(42);
  });
});
```

I add a file: `package.json`

```json
{
  "name": "demo",
  "scripts": {
    "test": "./node_modules/.bin/mocha tests/**"
  },
  "author": "@k33g aka @k33g_org",
  "devDependencies": {
    "chai": "^4.1.1",
    "mocha": "^3.5.0"
  }  
}
```

I created a build script

```shell
npm install
npm test 
```

I manually run the build: it's ok

- my default branch is `master`
- `master` is protected
- I checked `Require status checks to pass before merging`
- I checked `gitbucket-ci`

I created a new branch `test-1`

I added a new test to `tests/tests.js`

```javascript
describe('# 1 is 1', () => {
  it('should equal 1', () => {
    expect(1).to.equal(0);
  });
});
```

I created a new PR from `test-1`

<img width="859" alt="1" src="https://user-images.githubusercontent.com/491848/31479842-3e357474-af1a-11e7-80f9-9bf1ef6bc0b7.png">

I refreshed the page, the test failed, it's normal

<img width="857" alt="2" src="https://user-images.githubusercontent.com/491848/31479863-6027c776-af1a-11e7-913d-4442f8476804.png">

I fixed my test

```javascript
describe('# 1 is 1', () => {
  it('should equal 1', () => {
    expect(1).to.equal(1);
  });
});
```

I returned to the PR

the status is blocked on `pending check`

<img width="858" alt="3" src="https://user-images.githubusercontent.com/491848/31479876-7d2feb1e-af1a-11e7-832a-e8a2934f92ea.png">

**Comment by takezoe** [comment]
@k33g Thanks for detailed procesdure! I'll try again.

**Comment by takezoe** [comment]
@k33g I suppose maybe user who pushed the second commit does not exist on GitBucket. Build is kicked by commit hook, and to kick commit hooks, it requires that pusher is registered as same username in current implementation.

I'd like to fix this behavior in the future version of GitBucket, but could you test again after changing your git username to same as GitBucket username to confirm my hypothesis?

**Comment by k33g** [comment]
@takezoe for the tests, I don't use git, I directly update the code with the GitBucket UI (I know it's bad, but so easy)

**Comment by takezoe** [comment]
@k33g Oh, I see. It needs more investigation.

**Comment by takezoe** [comment]
@k33g I got it! Code editing in the browser doesn't kick the build because commit hooks are not applied to the online editing in GitBucket 4.17.0. This problem is going to be fixed in 4.18.0.

**Comment by k33g** [comment]
@takezoe Thank you for the answer 👍 
btw I'm very happy to see you at scala.io

**Comment by takezoe** [comment]
@k33g Me too! but we aren't good at English so we are working hard to prepare the presentation just now. 😢 
Anyway, GitBucket 4.18.0 and gitbucket-ci-plugin 1.1.0 has been released. Could you try again with them?
https://gitbucket.github.io/gitbucket-news/gitbucket/2017/10/14/gitbucket-4.18.0.html

**Comment by k33g** [comment]
@takezoe don't be afraid about 🇬🇧 mine is very creepy, you can ask https://github.com/ikeike443 ;)
🇫🇷 people don't speak 🇬🇧  so it's not a problem

I will test the new release 😄

**Comment by k33g** [comment]
@takezoe it works 👏 
I will play more with it this week end 😄

**Comment by takezoe** [comment]
@k33g That's good. So is it possible to close this issue?

and thanks for your encouragement. I know ikeda-san who is now GitHubber. He was Scala programmer and Play1 committer in the past. I'm sad a little because he hasn't been writing Scala for a long time. 😭

**Comment by k33g** [comment]
👍
